### PR TITLE
Add the /__version__ endpoint with the kinto or kinto-dist released version.

### PR DESCRIPTION
**Pull request by Natim**
Refs #591 

- [x] Add documentation.
- [x] Add tests.
- [x] Add a changelog entry.
- [x] If you changed the HTTP API, update the [API_VERSION](https://github.com/Kinto/kinto/blob/master/kinto/__init__.py#L15) constant and add an API changelog entry [in the docs](https://github.com/Kinto/kinto/blob/master/docs/api/index.rst)

r? @leplatrem

### CODE REVIEW

File: docs/api/index.rst [hunk 1]
```diff
@@ -11,6 +11,12 @@ API
 Changelog
 ---------
 
+1.9 (2016-07-29)
```
Comment (leplatrem) [inline docs/api/index.rst]: set to `unreleased` until it's released (unless we release 1.9 now)

File: CHANGELOG.rst [hunk 2]
```diff
@@ -16,6 +16,7 @@ This document describes changes between each past release.
 - Added ability to plug custom StatsD backend implementations via a new ``kinto.statsd_backend``
   setting. Useful for Datadog™ integration for example (fixes #626).
 - Added a ``delete-collection`` action to the ``kinto`` command. (#727)
+- Added a ``/__version__`` endpoint with the version that have been deployed. (#747)
```
Comment (leplatrem) [inline CHANGELOG.rst]: Mention it in a Protocol changes like other versions

File: docs/api/1.x/utilities.rst [hunk 3]
```diff
@@ -80,10 +80,30 @@ This endpoint is suitable for a load balancer membership test.
...
 
 The returned value is a JSON mapping containing open source contribution
 information as advocated by https://www.contributejson.org
+
+
+GET /__version__
+==================
+
+Return a JSON mapping containing information about what distribution
+have been deployed by OPS.
```
Comment (leplatrem) [inline docs/api/1.x/utilities.rst]: typo: has

File: docs/api/1.x/utilities.rst [hunk 4]
```diff
@@ -80,10 +80,30 @@ This endpoint is suitable for a load balancer membership test.
...
+
+::
+
+    {
+     "name":"kinto",
```
Comment (leplatrem) [inline docs/api/1.x/utilities.rst]: nitpick: identation

File: docs/api/1.x/utilities.rst [hunk 5]
```diff
@@ -80,10 +80,30 @@ This endpoint is suitable for a load balancer membership test.
...
+     "version":"3.3.2",
+     "commit":"ab8db089ee63dc8e14f4bcfc427a86f311dd7e52",
+     "source":"https://github.com/Kinto/kinto"
```
Comment (leplatrem) [inline docs/api/1.x/utilities.rst]: not: in shavar it was the `git remote` url I think

File: version.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -e
+
+cd "$(dirname "$0")"
+NAME=$(basename $PWD)
+SOURCE=$(git remote -v | grep origin | grep push | cut -f 2 | sed -e 's|git@|https://|g' | sed -e 's|github.com:|github.com/|g' | sed 's|.git (push)||g')
```
Comment (leplatrem) [inline version.sh]: Why not just `SOURCE = $(shell git config remote.origin.url)`

File: version.sh [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (leplatrem) [inline version.sh]: Why not in `Makefile` ?

Comment (Natim) [inline version.sh]: Oh we could do that yes. :+1:

**Comment by leplatrem** [comment]
I'm a bit suprised by the complexity of the view code, compared to https://github.com/mozilla-services/shavar/commit/cebd41f123d83d2198eef67d6a7da923873cd136

Also, consider:
* Update the release docs if there is anything new to do
* Reify the view result since it runs many commands that will never have different results within the same run

File: CHANGELOG.rst [hunk 8]
```diff
@@ -6,6 +6,10 @@ This document describes changes between each past release.
 3.4.0 (unreleased)
 ==================
 
+**Protocol**
+
+- Added a ``/__version__`` endpoint with the version that have been deployed. (#747)
```
Comment (leplatrem) [inline CHANGELOG.rst]: nit: «has»

File: kinto/core/views/version.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import pkg_resources
+
+from pyramid.security import NO_PERMISSION_REQUIRED
+from kinto.core import Service
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+ORIGIN = os.path.dirname(os.path.dirname(os.path.dirname(HERE)))
+
+VERSION_PATH = 'version.json'  # Relative to where we start it.
+VERSION_JSON = None
```
Comment (leplatrem) [inline kinto/core/views/version.py]: nitpick: move that to `except` ? (not sure)

**Comment by Natim** [comment]
The complexity comes from the fact that we have a fallback in case the version.json file is not present.

File: kinto/core/views/version.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    if VERSION_JSON is None:
+        settings = request.registry.settings
+        project_name = settings['project_name']
+
+        try:
+            # Try to get the project_name distribution
+            distribution = pkg_resources.get_distribution(project_name)
+        except pkg_resources.DistributionNotFound:
+            # If it doesn't work let's fallback to kinto
+            project_name = 'kinto'
```
Comment (leplatrem) [inline kinto/core/views/version.py]: project_name is always `kinto`. We use that for settings prefixes if I remember well

**Comment by Natim** [comment]
The major problem is that you cannot add the commit of the tag inside the file if you didn't commit yet. So it is impossible to actually release with the version.json file.

File: kinto/core/views/version.py (hunk 10 above)
Comment (Natim) [inline kinto/core/views/version.py]: No it's not, for syncto it is syncto for instance and readinglist, readinglist.

File: kinto/core/views/version.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+            distribution = pkg_resources.get_distribution(project_name)
+
+        package_version = distribution.version
+        package_metadata = distribution._get_metadata(distribution.PKG_INFO)
+
+        home_page = [m for m in package_metadata if m.startswith('Home-page:')]
+
+        source = None
+        if home_page:
+            source = home_page[0].split(':', 1)[1].strip()
```
Comment (leplatrem) [inline kinto/core/views/version.py]: This is weird, it would deserve some comments and some safety checks maybe

File: version.sh [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+set -e
+
+cd "$(dirname "$0")"
+NAME=$(basename $PWD)
+SOURCE=$(git config remote.origin.url | sed -e 's|git@|https://|g' | sed -e 's|github.com:|github.com/|g' | sed 's|\.git||g')
```
Comment (leplatrem) [inline version.sh]: Why don't you give the git remote url ?

Comment (leplatrem) [inline version.sh]: Oh I think I get it: ssh vs. https. Then no need to strip the trailing `.git` https://shavar.services.mozilla.com/__version__

**Pull request merged**